src/compute/aggregate/sum.rs

```diff
@@ -19,23 +19,43 @@ pub trait Sum<T> {
     fn simd_sum(self) -> T;
 }
 
+fn split_by_alignment<U, T>(values: &[T]) -> (&[T], &[T]) {
+    let alignment = std::mem::align_of::<U>();
+
+    let vals_ptr = values.as_ptr();
+    let bytes_offset = vals_ptr.align_offset(alignment);
+    let type_offset = if bytes_offset > 0 {
+        std::mem::align_of::<T>() / bytes_offset
+    } else {
+        0
+    };
+
+    let head = &values[..type_offset];
+    let aligned_values = &values[type_offset..];
+    (head, aligned_values)
+}
+
 #[multiversion]
 #[clone(target = "x86_64+avx")]
 fn nonnull_sum<T>(values: &[T]) -> T
 where
-    T: NativeType + Simd,
+    T: NativeType + Simd + Add<Output = T> + std::iter::Sum<T>,
     T::Simd: Add<Output = T::Simd> + Sum<T>,
 {
-    let mut chunks = values.chunks_exact(T::Simd::LANES);
+    let (head, aligned_values) = split_by_alignment::<T::Simd, _>(values);
+
+    let mut chunks = aligned_values.chunks_exact(T::Simd::LANES);
 
+    // Safety:
+    // we just made sure that we work on a slice af data aligned to T::Simd
     let sum = chunks.by_ref().fold(T::Simd::default(), |acc, chunk| {
-        acc + T::Simd::from_chunk(chunk)
+        acc + unsafe { T::Simd::from_chunk_aligned_unchecked(chunk) }
```

Comment:
we may not need the `unchecked` here (the compiler may figure it out; to test xD)